pkg/ruler/compat.go

```diff
@@ -233,17 +236,17 @@ func MetricsQueryFunc(qf rules.QueryFunc, queries, failedQueries prometheus.Coun
 			// All errors will still be counted towards "evaluation failures" metrics and logged by Prometheus Ruler,
 			// but we only want internal errors here.
 			if _, ok := querier.TranslateToPromqlAPIError(origErr).(promql.ErrStorage); ok {
-				failedQueries.Inc()
+				failedQueries.WithLabelValues(userID, failureReason).Inc()
 			}
 
 			// Return unwrapped error.
 			return result, origErr
 		} else if remoteQuerier {
-			// When remote querier enabled, consider anything an error except those with 4xx status code.
-			st, ok := grpcutil.ErrorToStatus(err)
-			if !(ok && st.Code()/100 == 4) {
-				failedQueries.Inc()
+			// When remote querier enabled, consider anything an "error" except those with 4xx status code.
+			if mimirpb.IsClientError(err) {
+				failureReason = "4xx"
```

Comment:
can we call them `client_error` and `server_error` instead, so they follow the same pattern?